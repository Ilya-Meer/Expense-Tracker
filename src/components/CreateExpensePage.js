import React, { Component } from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenseActions';

class CreateExpensePage extends Component {
  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }
  onSubmit(expense) {    
    this.props.addExpense(expense);
    this.props.history.push('/');
  }
  render() {
    return (
      <div>
        <p>I am the Create route!</p>
        <ExpenseForm onSubmit={this.onSubmit}/>
      </div>    
    )
  }
}

export { CreateExpensePage }


const mapDispatchToProps = (dispatch) => {
  return {
    addExpense: (expense) => {
      dispatch(addExpense(expense))
    }
  }
}
export default connect(undefined, mapDispatchToProps )(CreateExpensePage);