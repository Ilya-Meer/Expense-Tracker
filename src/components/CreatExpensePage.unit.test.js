import React from 'react';
import {shallow} from 'enzyme';
import CreateExpensePage from './CreateExpensePage';




describe('Create Expense Page', () => {
  it('renders properly', () => {
    expect(CreateExpensePage).toMatchSnapshot();
  })
})
