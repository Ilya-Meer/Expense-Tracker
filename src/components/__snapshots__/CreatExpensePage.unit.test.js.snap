// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Expense Page renders properly 1`] = `[Function]`;
